### PR TITLE
fix: set associatedWith to correct field with hasOne/hasMany and belongsTo

### PR DESCRIPTION
**Pull request by dpilch**
#### Description of changes

Fix regression when introduced with GQL v2 to set correct `associatedWith` field with `hasOne` and `belongsTo`.

#### Description of how you validated changes

QA Example:

```
type Project9 @model {
  id: ID!
  name: String
  team: Team9 @hasOne
}

type Team9 @model {
  id: ID!
  name: String!
  project: Project9 @belongsTo
}
```

Current incorrect generation:
```
// Incorrectly generated parent model
public final class Project9 implements Model {
  private final @ModelField(targetType="ID", isRequired = true) String id;
  private final @ModelField(targetType="String") String name;
  private final @ModelField(targetType="Team9") @HasOne(associatedWith = "id", type = Team9.class) Team9 team = null;
  private final @ModelField(targetType="ID") String project9TeamId;
}

// Child model is generated correctly, but I'll leave it here for reference
public final class Team9 implements Model {
  private final @ModelField(targetType="ID", isRequired = true) String id;
  private final @ModelField(targetType="String", isRequired = true) String name;
  private final @ModelField(targetType="Project9") @BelongsTo(targetName = "team9ProjectId", type = Project9.class) Project9 project;
}
```

Expected:
```
public final class Project9 implements Model {
  private final @ModelField(targetType="ID", isRequired = true) String id;
  private final @ModelField(targetType="String") String name;
  private final @ModelField(targetType="Team9") @HasOne(associatedWith = "project", type = Team9.class) Team9 team = null;
  private final @ModelField(targetType="ID") String project9TeamId;
}
```

I created a snapshot test to capture this.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-codegen/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] Breaking changes to existing customers are released behind a feature flag or major version update
- [ ] Changes are tested using sample applications for all relevant platforms (iOS/android/flutter/Javascript) that use the feature added/modified

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/appsync-modelgen-plugin/src/__tests__/visitors/gqlv2-regression-tests/__snapshots__/appsync-java-visitor.test.ts.snap [hunk 1]
```diff
@@ -1770,7 +1770,7 @@ public final class Post implements Model {
   public static final QueryField TITLE = field(\\"Post\\", \\"title\\");
   private final @ModelField(targetType=\\"ID\\", isRequired = true) String id;
   private final @ModelField(targetType=\\"String\\", isRequired = true) String title;
-  private final @ModelField(targetType=\\"Comment\\") @HasMany(associatedWith = \\"postCommentsId\\", type = Comment.class) List<Comment> comments = null;
```
Comment (phani-srikar) [inline packages/appsync-modelgen-plugin/src/__tests__/visitors/gqlv2-regression-tests/__snapshots__/appsync-java-visitor.test.ts.snap:1773]: schema for this test for easy reference:
```
      type Post @model {
        id: ID!
        title: String!
        comments: [Comment] @hasMany
      }

      type Comment @model {
        id: ID!
        content: String!
      }
```
Is the removal of `postCommentsId` field expected in this case?

File: packages/appsync-modelgen-plugin/src/__tests__/visitors/gqlv2-regression-tests/__snapshots__/appsync-java-visitor.test.ts.snap [hunk 2]
```diff
@@ -28,7 +28,7 @@ public final class Blog7V2 implements Model {
   public static final QueryField NAME = field(\\"Blog7V2\\", \\"name\\");
   private final @ModelField(targetType=\\"ID\\", isRequired = true) String id;
   private final @ModelField(targetType=\\"String\\", isRequired = true) String name;
-  private final @ModelField(targetType=\\"Post7V2\\") @HasMany(associatedWith = \\"blog7V2PostsId\\", type = Post7V2.class) List<Post7V2> posts = null;
+  private final @ModelField(targetType=\\"Post7V2\\") @HasMany(associatedWith = \\"blog\\", type = Post7V2.class) List<Post7V2> posts = null;
```
Comment (phani-srikar) [inline packages/appsync-modelgen-plugin/src/__tests__/visitors/gqlv2-regression-tests/__snapshots__/appsync-java-visitor.test.ts.snap:31]: schema for this test:
```
      type Blog7V2 @model {
        id: ID!
        name: String!
        posts: [Post7V2] @hasMany
      }
      type Post7V2 @model {
        id: ID!
        title: String!
        blog: Blog7V2 @belongsTo
        comments: [Comment7V2] @hasMany
      }
      type Comment7V2 @model {
        id: ID!
        content: String
        post: Post7V2 @belongsTo
      }
```

File: packages/appsync-modelgen-plugin/src/__tests__/visitors/gqlv2-regression-tests/__snapshots__/appsync-java-visitor.test.ts.snap [hunk 3]
```diff
@@ -616,7 +616,7 @@ public final class Project implements Model {
   public static final QueryField PROJECT_TEAM_ID = field(\\"Project\\", \\"projectTeamId\\");
   private final @ModelField(targetType=\\"ID\\", isRequired = true) String id;
   private final @ModelField(targetType=\\"String\\") String name;
-  private final @ModelField(targetType=\\"Team\\") @HasOne(associatedWith = \\"id\\", type = Team.class) Team team = null;
```
Comment (phani-srikar) [inline packages/appsync-modelgen-plugin/src/__tests__/visitors/gqlv2-regression-tests/__snapshots__/appsync-java-visitor.test.ts.snap:619]: relevant schema:
```
      # Implicit
      type Project @model {
        id: ID!
        name: String
        team: Team @hasOne
      }

      type Team @model {
        id: ID!
        name: String!
        project: Project @belongsTo
      }

      # Explicit
      type Project2 @model {
        id: ID!
        name: String
        team: Team2 @hasOne
      }

      type Team2 @model {
        id: ID!
        name: String!
        projectID: ID
        project: Project2 @belongsTo(fields: ["projectID"])
      }
```

**Review by phani-srikar — COMMENTED** [review]
Added relevant schemas to make it easier for the reviewers.

**Review by HuiSF — APPROVED** [review]
Changes look good to me. Also tested in a Flutter App, cascade delete with the `belongsTo` fix works as expected. `hasMany` remains the previous state (cascade delete not working).

**Review by iartemiev — APPROVED** [review]
The JS part LGTM 👍🏻

**Pull request merged**